python/lsst/drp/tasks/assemble_coadd.py

```diff
@@ -328,6 +336,9 @@ class AssembleCoaddTask(CoaddBaseTask, pipeBase.PipelineTask):
     ConfigClass = AssembleCoaddConfig
     _DefaultName = "assembleCoadd"
 
+    _doUsePsfMatchedPolygons: bool = False
+    """Use ValidPolygons from shrunk Psf-Matched Calexps?"""
+
```

Comment:
A safer way would be to make this an immutable property. However, I think marking this with a leading underscore is indication enough that this shouldn't be meddled with, while allowing for the possibility of toggling it for experimental reasons.

Reply:
I shall. But would you have felt the same way if I had defined this is as

```python
@property
@classmethod
def _doUsePsfMatchedPolygons(cls):
  return False
```

and the same with `True` in `CompareWarpAssembleCoaddTask` ?

Reply:
Well in that case you couldn't turn it on at all, so I guess any such note would only be useful if you wanted to advertise that it can be turned on in a derived class if... (whatever the conditions are).

Reply:
Oh, it's certainly mutable :-) You _could_ turn it on by setting
```python
AssembleCoaddTask._doUsePsfMatchedPolygons = property(lambda x: True)
```

I just don't expect anyone to turn it on and at least much less than anyone wanting to set `_doUsePsfMatchedPolygons`.

But this highlights the reason to move this away from `config`, which is that it should really be considered as a core part of the algorithm itself. The fact that this variable is needed is because of shared `run` method.

Reply:
I don't think that's actually mutating but reassigning, and hopefully we haven't merged any awful code like that without good reason.

Reply:
Then just make it a classmethod only, it's not that important tbat it be a property or attr.

Reply:
No, I think we agreed that leading underscore suffices.

Reply:
I thought you wanted it to be immutable, in which case it doesn't really matter if there's a leading underscore or not.

If you still want it to be mutable, fine, but if it's going to do the wrong thing when changed from the default then I'm not really sure what the benefit of that is.

Reply:
I just didn't want a naive user to go about changing the value, I don't care as much about the attribute itself being immutable. Being a `Config` field is an open invitation to change the value, which is now avoided. Leading underscore suggests that it _is possible_ to change the value at your own risk, but you _should not_ . And by you, I mean a user of this task, not a developer modifying this task.